scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -233,6 +239,7 @@
 	protected int statesIndex;
 	protected HandshakeState[] states;
 
+	protected boolean probe = false;
```

Comment:
Why adding all the "probe stuff" in Handshaker, it seems to me that this is only used in `ResumingClientHandshaker` ?

Reply:
If you search for the usage, you should find, that it's mainly used in places, where a general `Handshaker` is available. To add dummy implementations in the base and override them in the `ResumingClientHandshaker` seems for me to make it more complex. But possible ...

Reply:
Yep I saw that and "dummy implementations" was what I have in mind.
because this avoid to jump from class to class to understand the idea when reviewing (and so understanding the code)
(but this is not a strong opinion)